nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -340,6 +662,252 @@ impl DataStore {
+        let conn = self.pool_connection_unauthorized().await?;
+        Self::database_nexus_access_any_exist_on_connection(&conn).await
+    }
+
+    /// Checks if any db_metadata_nexus records exist in the database using an existing connection
+    pub async fn database_nexus_access_any_exist_on_connection(
+        conn: &async_bb8_diesel::Connection<DbConnection>,
+    ) -> Result<bool, Error> {
+        use nexus_db_schema::schema::db_metadata_nexus::dsl;
+
+        let exists: bool = diesel::select(diesel::dsl::exists(
+            dsl::db_metadata_nexus.select(dsl::nexus_id),
+        ))
+        .get_result_async(conn)
+        .await
+        .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(exists)
+    }
+
+    /// Registers a Nexus instance as having active access to the database
+    pub async fn database_nexus_access_insert(
+        &self,
+        nexus_id: OmicronZoneUuid,
+        state: DbMetadataNexusState,
+    ) -> Result<(), Error> {
+        use nexus_db_schema::schema::db_metadata_nexus::dsl;
+
+        let new_nexus = DbMetadataNexus::new(nexus_id, state);
+
+        diesel::insert_into(dsl::db_metadata_nexus)
+            .values(new_nexus)
+            .on_conflict(dsl::nexus_id)
+            .do_update()
+            .set(dsl::state.eq(excluded(dsl::state)))
+            .execute_async(&*self.pool_connection_unauthorized().await?)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(())
+    }
+
+    /// Registers multiple Nexus instances as having active access to the database
+    pub async fn database_nexus_access_insert_multiple(
+        &self,
+        nexus_ids: &[OmicronZoneUuid],
+        state: DbMetadataNexusState,
+    ) -> Result<(), Error> {
+        use nexus_db_schema::schema::db_metadata_nexus::dsl;
+
+        let new_nexuses: Vec<DbMetadataNexus> = nexus_ids
+            .iter()
+            .map(|&nexus_id| DbMetadataNexus::new(nexus_id, state))
+            .collect();
+
+        diesel::insert_into(dsl::db_metadata_nexus)
+            .values(new_nexuses)
+            .on_conflict(dsl::nexus_id)
+            .do_update()
+            .set(dsl::state.eq(excluded(dsl::state)))
+            .execute_async(&*self.pool_connection_unauthorized().await?)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(())
+    }
+
+    /// Initializes Nexus database access records from a blueprint using an
+    /// existing connection
+    ///
+    /// This function finds all Nexus zones in the given blueprint and creates
+    /// active database access records for them. Used during RSS rack setup.
+    ///
+    /// Returns an error if:
+    /// - Any db_metadata_nexus records already exist (should only be called
+    /// during initial setup)
+    pub async fn initialize_nexus_access_from_blueprint_on_connection(
```

Comment:
If we are okay existing in a period of time where `db_metadata_nexus` records *do not exist*, but blueprint execution could otherwise be functional, this change may not be necessary.

However, I think the presence of `active` records for live Nexuses acts as a strong guard against quiescing, as documented in https://rfd.shared.oxide.computer/rfd/588 , so they are populated here, within RSS setup.

Reply:
Yeah, I like the approach you've got here.